### PR TITLE
Added solution for problem #623

### PR DESCRIPTION
**Pull request by The-Divyansh**
Added new solution to ./solution/600-699

## Change Summary

Provide summary of changes with issue number if any.

## Checklist

**If you haven't fulfilled the below requirements, your PR won't be reviewed.**

### General 

- [x] Make sure you've read [CONTRIBUTING.md](https://github.com/wingkwong/leetcode-the-hard-way/blob/main/CONTRIBUTING.md) 
- [x] Make sure you've followed the template format
- [x] Make sure you've started the app locally and verified all the content and all links (if applicable) must be accessible correctly
- [x] Make sure you've applied LaTex for variables, formulas and time / space complexity instead of using backticks
- [x] Make sure you've formatted your code well (Code Block is rendered using `<Tabs/>`)
      Below is a C++ example. You can include multiple langs using `<TabItem/>` if necessary
      
````
<Tabs>

<TabItem value="cpp" label="C++">
<SolutionAuthor name="@YOUR_ALIAS"/>

```cpp
// Your code goes here
// Prefer K&R Coding style
```

</TabItem>

</Tabs>
````
- [x] Make sure you've explained your code well (write comments above each line - not inline)
- [x] Make sure you've written your explanation well and it is easy to understand for beginners
- [x] Make sure you've included Time & Space Complexity Analysis
- [x] Make sure you've included a meaningful approach name for your solution. Don't leave it blank. e.g. `## Approach 1: Two Pointers`.

### For Tutorial Articles

- [ ] Make sure you've explained your topic using 1 - 3 LC problems rather than using external problems. Solutions must be taken with explanations as well.
- [ ] Make sure you've provided suggested problems at the end using the given format. See [here](https://raw.githubusercontent.com/wingkwong/leetcode-the-hard-way/main/tutorials/math/number-theory/binary-exponentiation.md) as an example. If the target solution is not available, leave `solutionLink` blank.
- [ ] Make sure you've given credits / references if you use external resources. For an image, give credit under it. Otherwise, add a new section called References at the end (after Suggested Problems).

### CODE REVIEW

File: solutions/0600-0699/0623-add-one-row-to-tree-medium.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+We just run a dfs-like approach where we just follow the rules line-by-line as asked in the problem statement. Whenever we reach depth-1, we create two new nodes, and change the links.
+
+<Tabs>
+<TabItem value="cpp" label="C++">
+<SolutionAuthor name="@divyansh_0602"/>
+
+```cpp
+// Time Complexity: O(n). A total of n nodes of the given tree will be considered in worst case.
+// Space Complexity: O(N). The depth of the recursion tree can go upto n in the worst case(skewed tree).
```
Comment (wingkwong) [inline solutions/0600-0699/0623-add-one-row-to-tree-medium.md:61]: 1. `O(N)` -> `O(n)`
2. `worst case(skewed tree)` -> `worst case (skewed tree)`

**Review by wingkwong — CHANGES_REQUESTED** [review]

File: solutions/0600-0699/0623-add-one-row-to-tree-medium.md (hunk 1 above)
Comment (The-Divyansh) [reply solutions/0600-0699/0623-add-one-row-to-tree-medium.md]: Done

**Comment by The-Divyansh** [comment]
Updated required changes

File: solutions/0600-0699/0623-add-one-row-to-tree-medium.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+* `-100 <= Node.val <= 100`
+* `-10^5 <= val <= 10^5`
+* `1 <= depth <= the depth of tree + 1`
+
+## Approach 1: Recursion (DFS)
+
+We start DFS from root node and go into depth till we reach current depth = `depth-1`. Then we follow the rules given in problem statement i.e. create two new nodes, and insert these two new nodes between current node and its child node according to whether it is left or right child.
+
+When we reach `depth-1` and change the links, we do not go further into depth and just return from that DFS call.
+
```
Comment (wingkwong) [inline solutions/0600-0699/0623-add-one-row-to-tree-medium.md:56]: Still not detailed enough.

**Review by wingkwong — CHANGES_REQUESTED** [review]

**Comment by wingkwong** [comment]
Rejected. That's not written by you. Basically you copied the LC official explanation. (Even you've changed a few words, it's still not acceptable.)

**Pull request closed without merging**